### PR TITLE
portable sched_getcpu calls

### PR DESCRIPTION
**Pull request by ajkr**
- added a feature test in build_detect_platform to check whether sched_getcpu() is available. glibc offers it only on some platforms (e.g., linux but not mac); this way should be easier than maintaining a list of platforms on which it's available.
- refactored PhysicalCoreID() to be simpler / less repetitive. ordered the conditional compilation clauses from most-to-least preferred

### CODE REVIEW

File: build_tools/build_detect_platform [hunk 1]
```diff
@@ -401,6 +401,17 @@ EOF
...
+
+    # Test whether sched_getcpu is supported
+    $CXX $CFLAGS -x c++ - -o /dev/null 2>/dev/null  <<EOF
+      #include <sched.h>
+      int main() {
+        int cpuid = sched_getcpu();
+      }
+EOF
+    if [ "$?" = 0 ]; then
+        COMMON_FLAGS="$COMMON_FLAGS -DROCKSDB_SCHED_GETCPU_PRESENT"
```
Comment (siying) [inline build_tools/build_detect_platform]: Would you want to set it in build_tools/fbcode_config.sh and build_tools/fbcode_config4.8.1.sh too?

Comment (ajkr) [inline build_tools/build_detect_platform]: do we still use those? I noticed our TARGETS file manually sets some of these like -DROCKSDB_FALLOCATE_PRESENT. Is adding it to TARGETS sufficient?

Comment (ajkr) [inline build_tools/build_detect_platform]: oh never mind, I see why they're needed now.

Comment (siying) [inline build_tools/build_detect_platform]: And you have a good point. You need to add it to TARGETS file too. I think we may also missed several other flags there and TARGETS files.

File: build_tools/build_detect_platform [hunk 2]
```diff
@@ -401,6 +401,17 @@ EOF
     if [ "$?" = 0 ]; then
         COMMON_FLAGS="$COMMON_FLAGS -DROCKSDB_RANGESYNC_PRESENT"
```
Comment (siying) [inline build_tools/build_detect_platform]: Damn, we need to add this one too before the next release.

**Pull request closed without merging**